.prettierrc.json

```diff
@@ -0,0 +1,4 @@
+{
+  "trailingComma": "none",
```

Comment:
I find the trailing comma particularly useful in preventing accidental errors when extending a list ...

Reply:
Funny, I think the exact opposite :)  I think it's just as likely to cause errors because you cant see the end of the list; the only only place i've seen it before is in Dart code where it seems like a way to get around their "constructor hell".  I turned it off here so that it follows what we've done before (didnt it used to be an error in Javascript to have trailing commas?)

Reply:
I remember back to some problems with internet explorer not accepting trailing commas (I love them in perl!), especially with trailing commas of array elements (elisions).
I know that IE will be dead midth of this year, but I'm not sure about the implications allowing them.

Reply:
which means that I'm against allowing trailing commas :-)

Reply:
well, if you do a little google survey you will see, that as in may things perl has been way ahead of the curve ... in JS world, support for trailing (final) commas is being added in more and more places. So removing them or even disalowing them here seems to a rather anachronistic.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas

https://blog.logrocket.com/best-practices-using-trailing-commas-javascript/

https://time2hack.com/are-you-using-trailing-commas-in-your-javascript/

Reply:
but this isn't perl, and as the mozilla doc says, support has been added progressively, JSON does not support it at all, and it has inconsistencies between object literals and arrays.  The logrocket article shows how once is OK but you have to be careful not to add undefined entries in the array, IMHO meaning that allowing trailing commas means it's easier to add a bug.  The time2hack article finishes by saying that "We write code in Clean way for other developers to understand`, as if this somehow automatically qualifies as a higher grade of coding and the reader is stupid and inconsiderate and obfuscating to not just see it their way.

This is, of course, one of those arguments that something like prettier is supposed to resolve.  If we have to have commas at the end of lines then we can, but it just seems like another arbitrary change to something that has worked just fine for generations and which has just as many disadvantages as advantages.

Reply:
Not only Dart has trailing commas, 

Kotlin added them in 1.4 https://medium.com/@sanketmeghani/kotlin-1-4-trailing-commas-53c690eced32 

Go has them https://dave.cheney.net/2014/10/04/that-trailing-comma

Rust has them https://users.rust-lang.org/t/trailing-commas/13993

Trailing commas are a good thing -- They prevent a common bug when modifying existing code, and they make diffs shorter and more readable. 

So forcing them out of existance seems to be a rather odd move ...

Reply:
But I don't see that comparing it to other languages makes any sense; I havn't tried to find any, but do C++, C, etc?  Java does not.  Would we care if they did?  

> They prevent a common bug when modifying existing code

They also allow a common bug though ??

> So forcing them out of existance seems to be a rather odd move ...

Which we've never allowed before, so requiring them now is an equally odd move ...

Reply:
Just to add another vote, I agree with @johnspackman that trailing commas should be disallowed (removed by prettier).

Reply:
here is another one, but I guess your minds are made up ... 

https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8

but I guess rational arguments don't matter, because as one commenter puts it:

> I understand the benefits — or, I should say, the arguments in favour — but I still hate trailing commas.

I guess I can still configure es6ify to pretty up my own code with dangling commas, right ?

Reply:
I've had to write a lot of Dart/Flutter code over the last 9 months and having massively nested arrays and parameter lists is almost baked into the API; the default settings for Dart is to require a trailing comma and my experience is that always having commas makes it noticably harder to read (especially without an IDE to highlight the end of a statement, eg when browsing some code in Github).  Perhaps this is because a lot of Flutter devs are advocating the "style" which can lead to a single statement literally being hundreds of lines long, but I genuinely think that it does not have a net gain except a limited use case where there is a simple list and you want to mark & copy a horizontal line instead of a staggered one.

I understand the claimed benefits, I just think that they are not justified and don't outweigh the disadvantages.

> I guess I can still configure es6ify to pretty up my own code with dangling commas, right ?

Yes, this prettierrc.json only affects the framework; es6ify searches for the nearest prettierrc.json file, starting in the directory containing the .js file being prettier-ed and then going up through parent directories until it finds one.